src/models/voice_processing/whisper_transcriber_robust.py

```
@@ -0,0 +1,185 @@
+            import librosa
+            duration = librosa.get_duration(filename=str(audio_path))
+            return duration
+        except ImportError:
+            pass
+        except Exception as e:
+            logger.debug(f"Librosa duration extraction failed: {e}")
+
+        try:
+            # Try with ffprobe
+            import subprocess
+            result = subprocess.run([
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)